### DISCUSSION TITLE
how to create a neural network graph like netron?

### DISCUSSION THREAD

**Discussion by yanghw116** [zauberzeug/nicegui#2329] [Q&A]
### Description

hi everyone. I'm a rookie with web and gui coding. But I did well with python. So I'm glad to meet nicegui.

I need to visualize neural networks and do some modifications. Currently [netron](https://github.com/lutzroeder/netron) performs well in visualizing nets, but it lacks modifications. I read the code and it kills me with js and css. So, can anyone help me to create a graph as netron does? like this:

![image](https://github.com/zauberzeug/nicegui/assets/156150671/1978454a-3bfd-492c-9346-3388ac12776e)

**Comment by rodja** [comment]
Should be possible with https://nicegui.io/documentation/mermaid.

**Reply by yanghw116** [reply]
I tried, but I found it's much too simple. like this:

![image](https://github.com/zauberzeug/nicegui/assets/156150671/115e7585-eee3-49c2-bfc0-5b4939c51e5b)

can it be visually promoted?

**Reply by yanghw116** [reply]
I figured out that. Another problem is that it seems impossible to attach a recall function to the node in mermaid. Thus, graph modification with interaction seems to be impossible.
